src/services/Hiking.ts

```diff
@@ -0,0 +1,114 @@
+import axios, { AxiosResponse } from 'axios'
+import { ParsedQs, stringify as StringifyQs } from 'qs'
+import { DistanceRange, ResultsRange, Sort, StarsRange } from '../types/HikingTypes'
+import { config } from 'dotenv'
+import { response } from 'express'
+
+config()
+
+const API_KEY: string | undefined = process.env.HIKING_KEY
+
+interface GetTrailsByCordinatesProps {
+    /** 
+        * (required) `lat` - Latitude for a given area
+    */
+    lat: number
+    /** 
+        * (required) `lon` - Longitude for a given area
+     */
+    lon: number
+    /**
+        * (optional) `maxDistance` - Max distance, in miles, from lat, lon. Default: 30. Max: 200.
+    */
+    maxDistance?: DistanceRange
+    /**
+     * (optional) `maxResults` - Max number of trails to return. Default: 10. Max: 500.
+    */
+    maxResults?: ResultsRange
+    /**
+        * (optional) `sort` - Values can be 'quality', 'distance'. Default: quality.
+    */
+    sort?: Sort
+    /**
+        * (optional) `sort` - Min trail length, in miles. Default: 0 (no minimum).
+    */
+    minLength?: number
+    /*
+        * (optional) `minStars` - Min star rating, 0-4. Default: 0.
+    */
+    minStars?: StarsRange
+}
+
+const getTrailByIds = async (ids: string[] | number[]) => {
+    try {
+        if (!ids) throw Error('Trail id not provided.')
+
+        const uri = `https://www.hikingproject.com/data/get-trails-by-id?ids=${ids.join(',')}&key=${API_KEY}`
+
+        return await axios.get(uri)
+            .then((response: AxiosResponse) => {
+                return { success: true, data: response.data }
+            })
+            .catch(err => {
+                throw err
+            })
+    } catch (error) {
+        return { success: false, error: error.message }
+    }
+}
+
+const getTrailsByCordinates = async (queryString: ParsedQs) => {
+    try {
+        validateQueryString(queryString)
+
+        const uri = `https://www.hikingproject.com/data/get-trails?${StringifyQs(queryString)}&key=${API_KEY}`
+
+        return await axios.get(uri)
+            .then((response: AxiosResponse) => {
+                return { success: true, data: response.data }
+            })
+            .catch(err => {
+                throw err
+            })
+    } catch (error) {
+        return { success: false, error: error.message }
+    }
+}
+
+const getTrailConditions = async (ids: string[] | number[]) => {
+    try {
+        if (!ids) throw Error('Trail id not provided.')
```

Comment:
if you throw an error in the server, does it kill it? 
if so better to just return a custom error to the client for missing info and log it to winston logger (we need to add a logger)